javaslang/src/main/java/javaslang/collection/List.java

```diff
@@ -941,7 +941,7 @@ default T peek() {
         final Deque<T> preceding = new ArrayDeque<>(size());
         List<T> result = this;
         boolean found = false;
-        while (!found && !result.isEmpty()) {
+        while (!found && result.isNotEmpty()) {
```

Comment:
Generally I'm used to see `!someVar` or `!someMethodCall` and I would not have changed all occurreces of `!isEmpty` to `isNotEmpty`. `!` is part of the Java language and it is used everywhere. We can't live without it.

This line is a good example. We still have `!found`. We could rename it `notFound` and invert the content but that seems too complicated. We had to rewrite well-known algorithms to fit the new notation. Also it is not possible to eliminate `!` in algorithms that use both `someVar` and `!someVar`. So it is the best to accept `!`.

For the health of Javaslang it is better not to apply highly opinionated changes all over the code repository (a change may invent errors). It is like formatting. Some like 2 spaces indentation, some like 4 spaces or even tabs. Some prefer to use `!someVar`, some not.

Adding the additional method to `HashArrayMappedTrie` (HAMT) makes not much sense. We add code to an internal class just for a flavor of readability.

--

With this change we should add the method `isNotEmpty` to `Value` in order to offer the users of the _public_ API different ways to express themselves. But we should not change the whole code-base.